docs/programming_guide.rst

```diff
@@ -4,11 +4,31 @@
 Programming Guide - Developing Apps with NVIDIA FLARE
 #####################################################
 
-This guide contains details on the key concepts, objects, and information you should know to implement your own components.
+The NVIDIA FLARE Controller and Worker APIs are used to implement the task-based interaction defined in an :ref:`user_guide/application:NVIDIA FLARE Application`.
```

Comment:
should add general sections on 

* system architecture
    * *  concepts & system component
    * * events and event handles
* Federal Learning
* * Federated learning algorithm  -- fed avg, ...
* * Federated learning Algorithms -- CIFAIR10, ditto, Scaffolding etc.
* * Federal Analytics -- 

Again other sections can reference here by using links